starter-code/main.js

```diff
@@ -0,0 +1,296 @@
+              mainMenu()
+            });
+            break;
+          case "2":
+            // console.log('you typed 2');
+            db.collection('companies').count((error, result) => {
+              if (error) {
+                console.log(error);
+                rl.question(`\nType enter to continue: `, (answer) => {
+                  mainMenu()
+                });
+              } else {
+                console.log(result);
+                rl.question(`\nType enter to continue: `, (answer) => {
+                  mainMenu()
+                });
+              }
+            })
+            // rl.question(`\nType enter to continue: `, (answer) => {
+            //   mainMenu()
+            // });
+            break;
+
+          case "3":
+            db.collection('companies').find({
+              founded_year: 2004
+            }, {
+              name: 1,
+              _id: 0
+            }).count((error, result) => {
+              if (error) {
+                console.log(error);
+                rl.question(`\nType enter to continue: `, (answer) => {
+                  mainMenu()
+                });
+              } else {
+                console.log(result);
+                rl.question(`\nType enter to continue: `, (answer) => {
+                  mainMenu()
+                });
+              }
+            })
+
+            break;
+
+          case "4":
+            db.collection('companies').find({
+              founded_year: 2004,
+              founded_month: 2
+            }, {
+              name: 1,
+              _id: 0
+            }).toArray((error, result) => {
+              if (error) {
+                console.log(error);
+                rl.question(`\nType enter to continue: `, (answer) => {
+                  mainMenu()
+                });
+              } else {
+                console.log(result);
+                rl.question(`\nType enter to continue: `, (answer) => {
+                  mainMenu()
+                });
+              }
+            })
+            break;
+          case "5":
+            db.collection('companies').find({
+              founded_year: 2004,
+              founded_month: {
+                $gte: 4,
+                $lte: 6
+              }
+            }, {
+              name: 1,
+              _id: 0,
+              founded_year: 1,
+              founded_month: 1
+            }).sort({
+              founded_month: 1
+            }).toArray((error, result) => {
+              if (error) {
+                console.log(error);
+                rl.question(`\nType enter to continue: `, (answer) => {
+                  mainMenu()
+                });
+              } else {
+                console.log(result);
+                rl.question(`\nType enter to continue: `, (answer) => {
+                  mainMenu()
+                });
+              }
+            })
+            break;
+
+            // case "6":
```

Comment:
Mira una solución posible sería...
`
case "6":
            db.collection('companies').find({
              "offices.city": "Barcelona"
            }, {
              name: 1,
              _id: 0
            }).toArray((error, result) => {
              if (error) {
                console.log(error);
                rl.question(`\nType enter to continue: `, (answer) => {
                  mainMenu()
                });
              } else {
                console.log(result);
                rl.question(`\nType enter to continue: `, (answer) => {
                  mainMenu()
                });
              }
            })
            break;`